targets/esp8266/user/user_main.c

```diff
@@ -43,20 +42,19 @@ void show_free_mem(int idx) {
 
 //-----------------------------------------------------------------------------
 
-#include "jerry-targetjs.h"
-
-
 static int jerry_task_init(void) {
   int retcode;
   int src;
 
   DECLARE_JS_CODES;
 
+  js_entry();
+
   /* run main.js */
   show_free_mem(2);
-  retcode = js_entry(js_codes[0].source, js_codes[0].length);
+  retcode = js_eval(js_codes[0].source, js_codes[0].length);
   if (retcode != 0) {
-    printf("js_entry failed code(%d) [%s]\r\n", retcode, js_codes[0].name);
+    printf("js_eval failed code(%d) [%s]\r\n", retcode, js_codes[0].name);
```

Comment:
cannot this code simply be merged into the loop below? just start `src` from 0. oh, and yes: declare all the variables closest to their first use: `src` in the loop header, `retcode` in the loop body.

Reply:
Yeah, can be merged.